### DISCUSSION TITLE
Optimizing Training Efficiency by Running Checkpoint Saving in a Separate Thread

### DISCUSSION THREAD

**Discussion by binhphamthanh** [SWivid/F5-TTS#312] [Ideas]
Hello @SWivid and @lpscr,

In my experience with the training process, I’ve noticed that while the steps themselves run quite smoothly, the `save_checkpoint` phase takes a significant amount of time, which impacts the overall speed and duration of training.

Upon reviewing the code in trainer.py, specifically around line [293](https://github.com/SWivid/F5-TTS/blob/6d044f10b0f6b3a7b84844dfbc995382bc1a26b4/src/f5_tts/model/trainer.py#L293C21-L293C54), I’m considering whether we could run this part in a separate thread. This way, the `save_checkpoint` process could operate independently and would not require the training process to pause.

I haven't delved deeply into the code yet, but I'd like to know your thoughts on whether this approach would be feasible.

Thank you for your insights!

**Comment by SWivid** [comment]
Generally saving with larger interval will reduce the overhead which is fine.
